coordinator/src/main/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlanner.scala

```diff
@@ -33,6 +33,17 @@ class ShardKeyRegexPlanner(dataset: Dataset,
 
   override val schemas = Schemas(dataset.schema)
 
+  /**
+   * Returns true when regex has single matching value
+   * Example: sum(test1{_ws_ = "demo", _ns_ =~ "App-1"}) + sum(test2{_ws_ = "demo", _ns_ =~ "App-1"})
+   */
+  def hasSingleShardKeyMatch(nonMetricShardKeyFilters: Seq[Seq[ColumnFilter]]) = {
+    val shardKeyMatchers = nonMetricShardKeyFilters.map(shardKeyMatcher(_))
+    shardKeyMatchers.forall(_.size == 1) &&
+      shardKeyMatchers.forall(_.head.toSet.sameElements(shardKeyMatchers.head.head.toSet))
```

Comment:
@vishramachandran any concerns over this set conversion and comparison?

Reply:
I have concerns with `head.head` - susceptible to exception. Let us discuss.